### PR TITLE
Compile research.c as C++, small refactor

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2/management/Research.cpp [hunk 1]
```diff
@@ -611,24 +724,30 @@ void research_set_priority(sint32 activeCategories)
  *
  *  rct2: 0x00684A7F
  */
-void game_command_set_research_funding(sint32* eax, sint32* ebx, sint32* ecx, sint32* edx, sint32* esi, sint32* edi, sint32* ebp)
+void
+game_command_set_research_funding(sint32 * eax, sint32 * ebx, sint32 * ecx, sint32 * edx, sint32 * esi, sint32 * edi, sint32 * ebp)
```
Comment (janisozaur) [inline src/openrct2/management/Research.cpp]: that's a bit odd formatting here, the preferred form is to wrap arguments

**Comment by janisozaur** [comment]
Is the "small refactor" just introducing the constants?

File: src/openrct2/management/Research.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/management/Research.cpp]: Autoformatter :/

**Comment by Gymnasiast** [comment]
@janisozaur Pretty much, yes. Probably more of a clean-up than a proper refactor.

Anyway, fixed the formatting and rebased it.

**Pull request merged**